docs/csharp/whats-new/version-update-considerations.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Version and update considerations for C# developers
+description: Introducing new languages features in your library can impact the code that uses it.
+ms.date: 09/19/2018
+---
+
+# Version and update considerations for C# developers
+
+Compatibility is a very important goal as new features are added to the C# language. In almost all cases, you can use a new compiler version without breaking existing code.
+
+More care may be required when you adopt new language features in a library, and consumers of that library have not yet upgraded. The are two variations of compatibility: source compatible and binary compatible.
```

Comment:
Should we swap two definitions here? First - binary compatible, then - source compatible.
Rationale:
- If it's binary compatible, then it's source compatible. Maybe it's matter of taste; I would start with more strong compatibility.
- "More care may be required when you adopt new language features in a library, and consumers of that library have not yet upgraded" If consumers have not yet upgraded, then source compatibility is a no go. Consumers cannot rebuild, as they have not upgraded. So, better to aim to binary compatibility.
- The **Evaluate your library** section starts with discussion of binary compatibility, then goes on with source compatibility.

Reply:
> Should we swap two definitions here? First - binary compatible, then - source compatible.

I started writing them in that order. It came out very clumsy, so I switched. I'll try switching again in this round of edits.